classes/core/EntityDAO.inc.php

```diff
@@ -318,4 +319,20 @@ protected function convertToDB($value, string $type)
     {
         return $this->deprecatedDao->convertToDB($value, $type);
     }
+
+    /**
+     * Legacy function for grid handlers.
+     *Prepares LazyCollections in associative array GridHandler expects, e.g. [item_id => item]
+     *
+     * @see PKP\db\DAOResultFactory::toAssociativeArray()
+     *
+     */
+    public static function toAssociativeArray(LazyCollection $lazyCollection, string $idField = 'id'): array
+    {
+        $returner = [];
+        foreach ($lazyCollection as $item) {
+            $returner[$item->getData($idField)] = $item;
+        }
+        return $returner;
+    }
```

Comment:
See my comment above about moving this code to the `GridHandler`.

Reply:
Moved `toAssociativeArray ` to `GridHandler`